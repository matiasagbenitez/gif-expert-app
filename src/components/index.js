export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';
export * from './RemoveAllCategories';
export * from './RemoveCategory';