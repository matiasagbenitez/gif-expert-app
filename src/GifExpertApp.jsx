import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([
    "One Punchy",
    // "Samurai X",
    // "Dragon Ball",
  ]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory.toLowerCase())) return;
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory onNewCategory={(value) => onAddCategory(value)} />

      {categories.map((category) => {
        return <GifGrid key={category} category={category} />;
      })}
    </>
  );
};
