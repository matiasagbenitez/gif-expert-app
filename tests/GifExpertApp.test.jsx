import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Pruebas en el componente GifExpertApp", () => {
  test("Debe hacer match con el snapshot", () => {
    const wrapper = render(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
});
