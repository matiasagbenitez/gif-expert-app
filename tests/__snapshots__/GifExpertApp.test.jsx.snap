// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp Debe hacer match con el snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-5xl text-center font-bold text-slate-400"
      >
        GifExpertApp
      </h1>
      <p
        class="italic text-center text-slate-400"
      >
        Tu buscador de GIFs gratuito...
      </p>
      <form
        aria-label="form"
        class="bg-gray-400 rounded-t-lg pt-5 px-10 mt-5"
      >
        <p>
          <label
            class="text-md font-semibold"
          >
            Agregar categoría
          </label>
        </p>
        <input
          class="w-full my-5 p-2 border-2 border-gray-400 rounded-md focus:outline-none focus:ring-2 focus:ring-blue-600 focus:border-transparent bg-gray-200"
          placeholder="Ingrese una categoría..."
          type="text"
          value=""
        />
      </form>
      <div
        class="bg-gray-400 pb-5 px-10 last-of-type:rounded-b-lg"
      >
        <h2>
          Estos son los resultados de la búsqueda:
           
          <span
            class="font-semibold"
          >
            Las Vegas
          </span>
        </h2>
        <p>
          Loading...
        </p>
        <div
          class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-5 gap-5 my-5"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <h1
      class="text-5xl text-center font-bold text-slate-400"
    >
      GifExpertApp
    </h1>
    <p
      class="italic text-center text-slate-400"
    >
      Tu buscador de GIFs gratuito...
    </p>
    <form
      aria-label="form"
      class="bg-gray-400 rounded-t-lg pt-5 px-10 mt-5"
    >
      <p>
        <label
          class="text-md font-semibold"
        >
          Agregar categoría
        </label>
      </p>
      <input
        class="w-full my-5 p-2 border-2 border-gray-400 rounded-md focus:outline-none focus:ring-2 focus:ring-blue-600 focus:border-transparent bg-gray-200"
        placeholder="Ingrese una categoría..."
        type="text"
        value=""
      />
    </form>
    <div
      class="bg-gray-400 pb-5 px-10 last-of-type:rounded-b-lg"
    >
      <h2>
        Estos son los resultados de la búsqueda:
         
        <span
          class="font-semibold"
        >
          Las Vegas
        </span>
      </h2>
      <p>
        Loading...
      </p>
      <div
        class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-5 gap-5 my-5"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
